pages/cloudflare/bindings.mdx

```diff
@@ -0,0 +1,85 @@
+import { Callout } from 'nextra/components';
+
+### Bindings
+
+[Bindings](https://developers.cloudflare.com/workers/runtime-apis/bindings/) allow your Worker to interact with resources on the Cloudflare Developer Platform. When you declare a binding on your Worker, you grant it a specific capability, such as being able to read and write files to an [R2](https://developers.cloudflare.com/r2/) bucket.
+
+<Callout>
+If your app uses both bindings and @opennextjs/cloudflare, you currently cannot access bindings when running `next dev`. Instead, you must build your app using @opennextjs/cloudflare and then run it locally in the Workers runtime as described [here](/cloudflare/get-started). This means you cannot use hot-module reloading (HMR) for a faster development workflow. @opennextjs/cloudflare is pre 1.0, in active development, and we plan to address this soon.
+
+Alternatively, you can use [@cloudflare/next-on-pages](https://www.npmjs.com/package/@cloudflare/next-on-pages) to deploy Next.js apps to Cloudflare Pages. You can review the differences in supported Next.js features between @opennextjs/cloudflare and @cloudflare/next-on-pages [here](LINK).
+</Callout>
+
+#### How to configure your Next.js app so it can access bindings
+
+Install [@opennextjs/cloudflare](https://www.npmjs.com/package/@opennextjs/cloudflare), and then add a `wrangler.toml` file in the root directory of your Next.js app, as described in [Get Started](/cloudflare/get-started), y
+
+#### How to access bindings in your Next.js app
+
+You can access [bindings](https://developers.cloudflare.com/workers/runtime-apis/bindings/) from any route of your Next.js app via `getRequestContext`:
+
+```js
+// TODO: possible to bring this in from next-on-pages?
+import { getRequestContext } from "@cloudflare/next-on-pages";
+
+export async function GET(request) {
+	let responseText = "Hello World";
+
+	const myKv = getRequestContext().env.MY_KV_NAMESPACE;
+	await myKv.put("foo", "bar");
+	const foo = await myKv.get("foo");
+
+	return new Response(foo);
+}
+```
+
+#### How to add bindings to your Worker
+
+Add bindings to your Worker by [adding them to your `wrangler.toml` configuration file](/pages/functions/wrangler-configuration/).
+
+## TypeScript type declarations for bindings
+
+To ensure that the `env` object from `getRequestContext().env` above has accurate TypeScript types, install [`@cloudflare/workers-types`](https://www.npmjs.com/package/@cloudflare/workers-types) and create a [TypeScript declaration file](https://www.typescriptlang.org/docs/handbook/2/type-declarations.html).
+
+Install Workers Types:
+
+```sh
+npm install --save-dev @cloudflare/workers-types
+```
+
+Add Workers Types to your `tsconfig.json` file, replacing the date below with your project's [compatibility date](https://developers.cloudflare.com/workers/configuration/compatibility-dates/):
+
+```diff title="tsconfig.json"
+    "types": [
++        "@cloudflare/workers-types/2024-07-29"
```

Comment:
Do we want the new dynamic types here?

Reply:
We should discuss — I'm inclined to say yes since this is pre 1.0 and the person using it is already using something new and bleeding edge?